packages/web-config-server/src/apiV1/dataBuilders/generic/analytics/analyticsYearOnYear.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Tupaia Config Server
+ * Copyright (c) 2019 Beyond Essential Systems Pty Ltd
+ */
+
+import { formatLayeredDataElementCode, layerYearOnYear } from '../../../utils/layerYearOnYear';
+import { AnalyticsPerPeriodBuilder } from './analyticsPerPeriod';
+import { momentToPeriod, periodToMoment, reduceToDictionary } from '@tupaia/utils';
+import moment from 'moment';
+
+class AnalyticsYearOnYearBuilder extends AnalyticsPerPeriodBuilder {
+  async fetchAnalytics(
+    dataElementCodes,
+    additionalQueryConfig,
+    aggregationType = this.aggregationType,
+    aggregationConfig = this.config.aggregationConfig ?? {},
+  ) {
+    const originalStartDate = this.query.startDate;
+    const originalEndDate = this.query.endDate;
+
+    this.query.endDate = this.getTodayDataEndDate().format('YYYY-MM-DD');
+    this.query.startDate = this.getDataStartDate().format('YYYY-MM-DD');
```

Comment:
I ran the zoom functionality by @tcaiger and he seemed to be a fan of it and was happy for us to introduce it as a general feature across charts 🎉  I've created a card for it: https://github.com/beyondessential/tupaia-backlog/issues/2785

We don't necessarily need to implement it as part of this ticket, however when it does get implemented, there may need to be a couple of changes to this code:
1. No need to adjust the start/end dates of the query here
2. Revert changes to the date picker logic that have been implemented as part of this PR